src/store.js

```diff
@@ -1,29 +1,41 @@
-import Vue from 'vue'
-import Vuex from 'vuex'
+import Vue from 'vue';
+import Vuex from 'vuex';
 import { data } from './sample-data';
 
-Vue.use(Vuex)
+Vue.use(Vuex);
 
 const store = new Vuex.Store({
+  // strict: true,
   state: {
     todos: []
   },
   actions: {
     getSampleData(context) {
-      data.forEach( todo => {
-          context.commit('addTodo', todo);
-        }
-      )
+      data.forEach((todo) => {
+        context.commit('addTodo', todo);
+      });
     },
-    addTodo( context, newTodo ) {
+    addTodo(context, newTodo) {
       context.commit('addTodo', newTodo);
+    },
+    updateTodo(context, todo) {
+      context.commit('updateTodo', todo);
     }
-  },  
+  },
   mutations: {
     addTodo(state, newTodo) {
-      state.todos.push(newTodo)
+      state.todos.push(newTodo);
+    },
+    updateTodo(state, modifiedTodo) {
+      const index = state.todos.findIndex((todo) => {
+        return todo.title === modifiedTodo.title && todo.notes === modifiedTodo.notes;
+      });
+      state.todos.splice(index, 1, modifiedTodo);
+      // https://vuejs.org/v2/guide/list.html#Caveats
+      // state.todos[index] = modifiedTodo;
+      // state.todos[index] = Object.assign({}, modifiedTodo);
```

Comment:
The caveat I mentioned, we should strip these out but I wanted to show what I tried and the url to the docs.